apps/annotations/models.py

```diff
@@ -0,0 +1,64 @@
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+from django.utils.translation import pgettext_lazy
+from field_audit import audit_fields
+from taggit.managers import TaggableManager
+from taggit.models import TagBase, TaggedItem
+
+from apps.teams.models import BaseTeamModel, Team
+from apps.users.models import CustomUser
+
+
+@audit_fields(
+    "name",
+    "slug",
+    "created_by",
+    "team",
+)
+class Tag(TagBase, BaseTeamModel):
+    name = models.CharField(verbose_name=pgettext_lazy("A tag name", "name"), max_length=100)
+    created_by = models.ForeignKey("users.CustomUser", on_delete=models.DO_NOTHING)
+
+    class Meta:
+        verbose_name = _("Tag")
+        verbose_name_plural = _("Tags")
+        unique_together = ("team", "name")
+        ordering = ["name"]
+
+
+@audit_fields(
```

Comment:
Do you know how easy it is to query these? e.g. if we were showing a list of changes to a 'chat' object and wanted to include changes to tags?

Reply:
Easy
```python
from field_audit.models import AuditEvent
AuditEvent.objects.filter(...)
```

Reply:
Sure, but I mean if I give you a chat object and and want to get back a list of tag changes for that chat?

Reply:
Right. So this is what we have from the audit log:

![image](https://github.com/dimagi/open-chat-studio/assets/64970009/574b3e82-886e-4e22-8b9f-dd7e9917e641)

We can use the object's app + model name to query the audit event records, then we can pluck the username from the `change_context` and find the user that did it (or just look at `delta` if we only care about what changed).